File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.js
'use strict';
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
let input = require('fs').readFileSync(path).toString().trim().split('\n');

const [queueInfo, idxArr] = input;

const [N, M] = queueInfo.split(' ').map(Number);
const numIndexArr = idxArr.split(' ').map(Number);
const ringBuffer = Array.from({ length: N }).map((_, i) => i + 1);
let result = 0;

for (let num of numIndexArr) {
  if (ringBuffer[0] === num) {
    ringBuffer.shift();
    // num 건너뛰기
    continue;
  } else if (ringBuffer.indexOf(num) > ringBuffer.length / 2) {
    while (ringBuffer[0] !== num) {
      ringBuffer.unshift(ringBuffer.pop());
      result++;
    }
    ringBuffer.shift();
  } else {
    while (ringBuffer[0] !== num) {
      ringBuffer.push(ringBuffer.shift());
      result++;
    }
    ringBuffer.shift();
  }
}

console.log(result);

File: 백준/Silver/1021. 회전하는 큐/회전하는 큐.py
import sys
from collections import deque 

input = sys.stdin.readline

N, M = map(int, input().strip().split())
target_nums = list(map(int, input().split()))

result = 0
dequeue = deque([i + 1 for i in range(N)])


for num in target_nums:
  num_idx = dequeue.index(num)
  middle_idx = len(dequeue) // 2

  if num_idx <= middle_idx:
    for _ in range(num_idx):
      result += 1
      dequeue.append(dequeue.popleft())
  else:
    for _ in range(len(dequeue) - num_idx):
      result += 1
      dequeue.appendleft(dequeue.pop())

  dequeue.popleft()
  
print(result)